pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -168,37 +172,31 @@ public void removeSegmentsFromStore(String tableNameWithType, List<String> segme
 
   protected void removeSegmentFromStore(String tableNameWithType, String segmentId) {
     final String rawTableName = TableNameBuilder.extractRawTableName(tableNameWithType);
-    if (_localDiskDir != null) {
-      File fileToMove = new File(new File(_localDiskDir, rawTableName), segmentId);
-      if (fileToMove.exists()) {
-        File targetDir = new File(new File(_localDiskDir, DELETED_SEGMENTS), rawTableName);
-        try {
+    if (_dataDir != null) {
+      URI deletedSegmentDirURI;
+      URI fileToMoveURI;
+      PinotFS pinotFS;
+      URI dataDirURI = ControllerConf.getUriFromPath(_dataDir);
+      fileToMoveURI = ControllerConf.constructSegmentLocation(_dataDir, rawTableName, segmentId);
+      pinotFS = PinotFSFactory.create(dataDirURI.getScheme());
+      deletedSegmentDirURI =
+          ControllerConf.getUriFromPath(StringUtil.join(File.separator, _dataDir, DELETED_SEGMENTS, rawTableName));
+      try {
+        if (pinotFS.exists(fileToMoveURI)) {
           // Overwrites the file if it already exists in the target directory.
-          FileUtils.copyFileToDirectory(fileToMove, targetDir, false);
-          LOGGER.info("Moved segment {} from {} to {}", segmentId, fileToMove.getAbsolutePath(), targetDir.getAbsolutePath());
-          if (!fileToMove.delete()) {
-            LOGGER.warn("Could not delete file", segmentId, fileToMove.getAbsolutePath());
+          pinotFS.move(fileToMoveURI, deletedSegmentDirURI);
```

Comment:
Don't we need to construct a file URI to move to?  So, we need to include the segmentId in line 183?

Reply:
good catch, thank you